sandbox/next/package.json

```diff
@@ -9,18 +9,24 @@
     "lint": "next lint"
   },
   "dependencies": {
-    "next": "14.2.15",
-    "react": "^18.3.1",
-    "react-dom": "^18.3.1",
+    "next": "15.0.1",
+    "react": "19.0.0-rc-69d4b800-20241021",
+    "react-dom": "19.0.0-rc-69d4b800-20241021",
     "smarthr-ui": "workspace:*"
   },
   "devDependencies": {
     "@playwright/test": "^1.48.2",
     "@types/node": "^20.17.1",
-    "@types/react": "^18.3.12",
-    "@types/react-dom": "^18.3.1",
+    "@types/react": "npm:types-react@19.0.0-rc.1",
+    "@types/react-dom": "npm:types-react-dom@19.0.0-rc.1",
     "postcss": "^8.4.47",
     "tailwindcss": "^3.4.14",
     "typescript": "^5.6.3"
+  },
+  "pnpm": {
+    "overrides": {
+      "@types/react": "npm:types-react@19.0.0-rc.1",
+      "@types/react-dom": "npm:types-react-dom@19.0.0-rc.1"
+    }
```

Comment:
codemodの実行結果でpackage.jsonの中身が変わってます。
一応、React v19を使わないというのもcodemod上で選択できましたが、smarthr-uiとしてはReactの最新バージョンに追従していくのと、下記のように動作も安定していると強気のコメントを頂いているのでReact v19でcodemodを実行してます。

>Although React 19 is still in the RC phase, our extensive testing across real-world applications and our close work with the React team have given us confidence in its stability. The core breaking changes have been well-tested and won't affect existing App Router users. Therefore, we've decided to release Next.js 15 as stable now, so your projects are fully prepared for React 19 GA.

https://nextjs.org/blog/next-15#react-19

Reply:
今は sandbox は RSC 描画可能かを判定することで使ってるので、RSC 周りがバージョンで挙動変わると嫌だなぁと感じてます。

Next はもうすぐにでも15使いそうですが、React はプロダクト側が 19 にする日がまだ結構先だと思うので、主要バージョンの18を使ってテスト出来る状態にしておくのが適切かなと思いましたがどうでしょうか。

Reply:
>React v19を使わないというのもcodemod上で選択できましたが

なんか強制的にv19になりますね...

```
next % npx @next/codemod@latest upgrade latest
  Detected installed versions:
  - React: v18.3.1
  - Next.js: v15.0.1
✔ Enable Turbopack for next dev? … no
✔ Would you like to run the React 19 upgrade codemod? … no
✔ Would you like to run the React 19 Types upgrade codemod? … no
Upgrading your project to Next.js 15.0.2...

 WARN  The field "pnpm.overrides" was found in /Users/hiroki.miyaji/git/smarthr-ui/sandbox/next/package.json. This will not take effect. You should configure "pnpm.overrides" at the root of the workspace instead.
Scope: all 3 workspace projects
```

以下codemodの結果

```
   "dependencies": {
-    "next": "15.0.1",
-    "react": "^18.3.1",
-    "react-dom": "^18.3.1",
+    "next": "15.0.2",
+    "react": "19.0.0-rc-02c0e824-20241028",
+    "react-dom": "19.0.0-rc-02c0e824-20241028",
     "smarthr-ui": "workspace:*"
   },
   "devDependencies": {
     "@playwright/test": "^1.48.2",
     "@types/node": "^20.17.1",
-    "@types/react": "^18.3.12",
-    "@types/react-dom": "^18.3.1",
+    "@types/react": "npm:types-react@19.0.0-rc.1",
+    "@types/react-dom": "npm:types-react-dom@19.0.0-rc.1",
     "postcss": "^8.4.47",
     "tailwindcss": "^3.4.14",
     "typescript": "^5.6.3"
+  },
+  "pnpm": {
+    "overrides": {
+      "@types/react": "npm:types-react@19.0.0-rc.1",
+      "@types/react-dom": "npm:types-react-dom@19.0.0-rc.1"
+    }
   }
 }
```

Reply:
@s-sasaki-0529 @moshisora 
(とりあえずレビュアーのお二人にメンション飛ばします)
私のググり力がないかもなんですが、Next.js v15 と React v18 の互換性に関する情報が見つかりませんでした。
一旦安牌な選択肢取って Next.js v15 に上げずに Next.js v14 ＆ React v18 のまま（＝このPRは閉じる）にしようと思うのですがいかがでしょうか？

Reply:
ほえ〜、実質 React v19 を使って当たり前みたいな感じなんですかねー。

普通に社内で Next.js (特に AppRouter でサーバーコンポーネント) 使ってるプロダクトも、Next 15 にあげるときに React v19 にしてるんでしょうか。であればもうそっちの多数派に合わせても良い気もしてきました。

一旦閉じて様子見でもOKです！必要なときにこのPRを元にまたバージョンアップすれば良さそうなので！

Reply:
では一旦閉じちゃいます！